prompt_poet/template.py

```diff
@@ -23,6 +25,7 @@ class Template:
     def __init__(
         self,
         template_path: str = None,
+        template_loader: TemplateLoader = None,
         package_name: str = None,
```

Comment:
can you add template_loader to the docstring and mark template_path and package_name as "deprecated" and log a warning if they are not None. we can remove them in a later pr. thanks!

Reply:
done

Reply:
also marked template_name and template_dir deprecated, as with template_loader these fields are not necessary there.